src/simple_lottery/init.rs

```diff
@@ -0,0 +1,94 @@
+//! Parameter derivation for simple lottery.
+
+use super::params::Params;
+use super::setup::Setup;
+
+use std::f64::consts::LN_2;
+
+/// Calculates Setup parameters.
+pub fn make_setup(params: &Params) -> Setup {
+    let ratio = params.set_size as f64 / params.lower_bound as f64;
+
+    // Execute binary search for the optimal value of `rs`.
```

Comment:
It is much more efficient though. 
Yes, we shouldn't run the setup many times, and it can be done beforehand, but I do not think adding a 6 line long block would make the code less readable or harder to maintain.

Reply:
It would add an extra code path with no clear benefit. I'm against extra complexity until it is justified by real requirements.

Reply:
There is a clear benefit. We would not have to go through a loop and directly output the result. We would also be closer to the paper, which always is helpful for a reference implementation.

Reply:
> We would not have to go through a loop and directly output the result.

And what would be the benefit of that?